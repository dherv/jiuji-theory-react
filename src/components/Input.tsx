import styled from "styled-components";
import { themeBlueInput } from "../styled/themes";

const Input = styled.input`
  padding: 0.25rem 1rem;
  border: ${(props) => props.theme.border};
  font-weight: 500;
  font-size: 12px;
`;

Input.defaultProps = {
  theme: themeBlueInput,
};

export default Input;
