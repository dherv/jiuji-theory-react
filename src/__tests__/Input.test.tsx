import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Input from "../components/Input";

const props = {
  onChange: jest.fn(),
  placeholder: "test",
};

test("should display an Input with a placeholder", () => {
  render(<Input {...props} />);
  expect(screen.getByPlaceholderText("test")).toBeInTheDocument();
});

test("should call onChange when user type in the input", () => {
  render(<Input {...props} />);
  fireEvent.change(screen.getByPlaceholderText("test"), {
    target: { value: "23" },
  });
  expect(props.onChange).toHaveBeenCalled();
});

test("should render the Input correctly", () => {
  const { asFragment } = render(<Input {...props} />);
  expect(asFragment()).toMatchSnapshot();
});
