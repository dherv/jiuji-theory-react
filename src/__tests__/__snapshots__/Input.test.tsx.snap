// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Input correctly 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm ldTfNc"
    placeholder="test"
  />
</DocumentFragment>
`;
